src/toil/lib/aws/iam.py

```diff
@@ -88,7 +120,49 @@ def check_permission_allowed(perm: str, list_perms: List[str]) -> bool:
             return True
     return False
 
-def allowed_actions_roles(iam: IAMClient, policy_names: List[str], role_name: str) -> Dict[str, List[str]]:
+def get_actions_from_policy_document(policy_doc: Dict[str, Any]) -> AllowedActionCollection:
+    '''
+    Given a policy document, go through each statement and add actions and not actions to an action collection
```

Comment:
Maybe we could call the concept of "actions and not actions" something like "permission grants"?

Or we could say:
```suggestion
    Given a policy document, create an AllowedActionCollection describing the permissions it grants.
```